### PR TITLE
lib: Remove dirent64-related build warnings

### PR DESCRIPTION
**Pull request by TAGOliveira201000**
Using Musl and the dirent64 structure causes build warnings. This is because certain source code files do not define _GNU_SOURCE. Fix build warnings by defining _GNU_SOURCE in the required source code files

### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): any application using Musl

### CODE REVIEW

File: lib/posix-socket/driver.c
Comment (StefanJum) [inline lib/posix-socket/driver.c:36]:
```suggestion
#define _GNU_SOURCE

```

**Comment by razvand** [comment]
This is OK with me. I'll wait for @StefanJum to provide the approval message, and then I'll do the final approval step.

**Review by StefanJum — CHANGES_REQUESTED** [review]
One comment, all good besides that.

**Review by StefanJum — APPROVED** [review]
All good now.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**
